cpp/src/arrow/python/numpy_to_arrow.cc

```diff
@@ -743,7 +743,9 @@ Status NumPyConverter::ConvertDecimals() {
 
   if (type_ == NULLPTR) {
     for (PyObject* object : objects) {
-      RETURN_NOT_OK(max_decimal_metadata.Update(object));
+      if (!internal::PandasObjectIsNull(object)) {
+        RETURN_NOT_OK(max_decimal_metadata.Update(object));
+      }
     }
 
     type_ =
```

Comment:
By the way, what happens here if all items are None? Do we have a test for that?

Reply:
I'll add that

Reply:
done